salalib/alllinemap.cpp

```diff
@@ -172,39 +172,43 @@ std::tuple<std::unique_ptr<ShapeGraph>, std::unique_ptr<ShapeGraph>> AllLineMap:
    // a little further setting up is still required...
    std::map<RadialKey,RadialSegment> radialsegs;
 
-   // now make radial segments from the radial lines... (note, start at 1)
-   auto iter = m_radial_lines.begin();
-   auto prevIter = m_radial_lines.begin();
-   ++iter;
-   for (;iter != m_radial_lines.end();) {
-      if (iter->vertex == prevIter->vertex && iter->ang != prevIter->ang) {
-                radialsegs.insert(std::make_pair( (RadialKey)(*iter), (RadialSegment)(*prevIter)));
-      }
-      ++iter;
-      ++prevIter;
+   if(m_radial_lines.size() > 0) {
+       // now make radial segments from the radial lines... (note, start at 1)
+       auto iter = m_radial_lines.begin();
```

Comment:
Would be cleaner by using `if ( iter != m_radial_lines.end()){` rather than using size ().
Or have flag `bool first` and write a standard loop, on the first iteration just assigning `prev_iter = iter;`

Reply:
Fair point, fixed with first solution